exercise/static/exercise/assessment.js

```diff
@@ -32,7 +36,29 @@ $(function () {
           const text = $("<pre/>").text(data);
           text.attr('data-url', fileUrl);
           element.find('.submitted-file-data').html(text);
-          text.highlightCode();
+          var isSticky = localStorage.getItem('submissionSticky') === 'true';
+          if (isSticky) {
+            element.addClass('sticky');
+          }
+          var stickyButton = {
+            action: function() {
+              isSticky = !isSticky;
+              element.toggleClass('sticky', isSticky);
+              localStorage.setItem('submissionSticky', isSticky);
+            },
+            icon: isSticky ? 'check' : 'unchecked',
+            text: _('Scroll separately'),
+            toggle: true,
+          };
+          text.highlightCode({extraButtons: [stickyButton]});
+
+          if (canSticky) {
+            const panelHeading = $('.assessment-panel .panel-heading');
+            const observer = new ResizeObserver(function () {
+              element.parent().css('--sticky-top', panelHeading.outerHeight() + 'px');
+            });
+            observer.observe(panelHeading.get(0));
+          }
```

Comment:
This code is similar to the code below that is used for grader feedback. Can we make it into a function?